SOP/SOP-push.adoc

```diff
@@ -0,0 +1,179 @@
+
+=== UnifiedPushApiHighRequestFailure
+
+- Check <<General procedure>>
+
+=== UnifiedPushPodCPUHigh
+
+- Check <<General procedure>>
+
+=== UnifiedPushJavaHeapThresholdExceeded
+
+- Check <<General procedure>>
+
+=== UnifiedPushJavaNonHeapThresholdExceeded
+
+- Check <<General procedure>>
+
+=== UnifiedPushJavaGCTimePerMinuteScavenge
+
+- Check <<General procedure>>
+
+== Warning
+
+=== UnifiedPushMessagesFailures
+
+- Check <<General procedure>>
+
+=== UnifiedPushPodMemoryHigh
+
+- Check <<General procedure>>
+
+=== UnifiedPushApiHighRequestDuration
+
+- Check <<General procedure>>
+
+=== UnifiedPushApiHighConcurrentRequests
+
+- Check <<General procedure>>
+
+=== UnifiedPushJavaDeadlockedThreads,
+
+- Check <<General procedure>>
+
+== General procedure
+
+. Capture a snapshot of the 'UnifiedPush Server' Grafana dashboard and track it over time. The metrics can be useful for identifying performance issues over time.
+
+Troubleshoot using the following steps via either the console and/or the cli:
+
+== Console:
+
+. Capture application logs for analysis.
+.. Go to : `Applications -> Pods -> example-unifiedpushserver-<xyz123> -> Logs -> Container -> ups`
+.. Go to : `Applications -> Pods -> example-unifiedpushserver-<xyz123> -> Logs -> Container -> ups-oauth-proxy`
+.. Go to : `Applications -> Pods -> example-unifiedpushserver-postgresq-<xyz123> -> Logs -> Container -> postgresq`
+. If necessary, recreate the service pod to restore service.
+.. Navigate to `Application -> Pods -> example-unifiedpushserver-<xyz123> -> Actions -> Delete -> Delete`
+.. Navigate to `Application -> Pods -> example-unifiedpushserver-postgresq-<xyz123> -> Actions -> Delete -> Delete`
+. Check the operator pod is present as it is responsible for managing the service pod as described in https://github.com/aerogear/unifiedpush-operator/blob/0.1.0/SOP/SOP-operator.adoc[UnifiedPushOperatorDown]
+
+== CLI:
+
+. Capture application logs for analysis.
+.. Capture the logs of the UPS Container
+... Get the service pod name -> `oc get pods | grep unifiedpushserver`
+... Run `oc logs <service-podname> -c ups`
+... Save the logs by running `oc logs <service-podname> -c ups > <filename>.log`
+.. Capture the logs of the UPS OAuth Proxy Container
+... Get the service pod name -> `oc get pods | grep unifiedpushserver`
+... Run `oc logs <service-podname> -c ups-oauth-proxy`
+... Save the logs by running `oc logs <service-podname> -c ups-oauth-proxy > <filename>.log`
+.. Capture the logs of the UPS Database
+... Get the service pod name -> `oc get pods | grep unifiedpushserver-postgresq`
+... Run `oc logs <service-podname> -c unifiedpushserver-postgresq`
+... Save the logs by running `oc logs <service-podname> -c unifiedpushserver-postgresq > <filename>.log
+. If necessary, recreate the service pod to restore service.
```

Comment:
if it's performance issue, we should be able to increase the number of UPS pods?

Reply:
Also for possible performance issues, we should:

1. Check the performance of the database pod (CPU, memory usage) and logs to see if the database is under any pressure.
2. Check the number of pending messages on the queue (is it too high?) Are we seeing the number of pending messages keep growing?
3. Is UPS server itself is under pressure? If so try scale ups itself by increasing the number of pods.

Reply:
@wei-lee what are the steps to do it with the UPS OPer? (You can check it installed here https://master.london-4bc4.openshiftworkshop.com/console/project/unifiedpush/overview)

It is using the Deployment Config? Also, is not the size specified in the CR? @grdryn

Reply:
@wei-lee @grdryn explained that is not possible to scale.

Reply:
it will be possible to scale UPS very soon.

Reply:
So, when it be possible we should change here or would you like that I added steps to change the size spec in the CR besides it do not exist yet? @wei-lee @grdryn